packages/app/src/providers/utils.ts

```diff
@@ -6,32 +6,14 @@ import type {
   ScalarShape,
 } from '@h5web/shared/hdf5-models';
 import { DTypeClass } from '@h5web/shared/hdf5-models';
+import type { OnProgress } from '@h5web/shared/react-suspense-fetch';
+import type { AxiosProgressEvent } from 'axios';
 import { AxiosError } from 'axios';
 
 import type { DataProviderApi } from './api';
 
 export const CANCELLED_ERROR_MSG = 'Request cancelled';
 
-export async function handleAxiosError<T>(
-  func: () => Promise<T>,
-  getErrorToThrow: (status: number, errorData: unknown) => string | undefined,
-): Promise<T> {
-  try {
-    return await func();
-  } catch (error) {
-    if (error instanceof AxiosError && error.response) {
-      const { status, data } = error.response;
-      const errorToThrow = getErrorToThrow(status, data);
-
-      if (errorToThrow) {
-        throw new Error(errorToThrow);
-      }
-    }
-
-    throw error;
-  }
-}
```

Comment:
This utility function was not very elegant to use. I replaced it with `processAxiosError` below, which basically contain what was inside the try/catch block of `handleAxiosError`.

Another change, however, is that `getMessageToThrow` receives the whole `AxiosError` instead of just the response. This allows using `processAxiosError` to detect and process "cancelled" errors (which don't have a response since they're cancelled).